pieces/rest_api_postgres/rust/src/main.rs

```diff
@@ -0,0 +1,315 @@
+mod db;
+mod models;
+mod routes;
+
+use axum::Router;
+use dotenvy::dotenv;
+
+use crate::db::create_pool;
+
+#[tokio::main]
+async fn main() {
+    dotenv().ok();
+    let pg_uri = dotenvy::var("DATABASE_URL").expect("Invalid DB URI");
+    let app = app(&pg_uri).await;
+
+    let listener = tokio::net::TcpListener::bind("0.0.0.0:3000").await.unwrap();
+    axum::serve(listener, app).await.unwrap();
+}
+
+async fn app(pg_uri: &str) -> Router {
+    let pool = create_pool(pg_uri)
+        .await
+        .expect("Unable to connect to the database");
+
+    Router::new()
+        .merge(routes::book::book_routes(pool.clone()))
```

Comment:
Is there a reason we're not doing `Arc::clone(&pool)` and using a direct clone instead? `Arc` seemed to do pretty well in terms of performance for the ETL queries I ran in [the other piece](https://github.com/thedataquarry/rustinpieces/blob/main/pieces/postgres_etl/rust/src/main.rs), so maybe this could help improve performance in your benchmark?

Reply:
I pulled this straight from the Axum examples. I figured there must be a reason they did it this way, but I can try an `Arc`.

Reply:
I tried using `Arc` and it let me do that, but on the other end it made me remove it from the `Arc` and clone it there. So in it ended up forcing this way of doing it. This must be why the example does it this way.